### ISSUE TITLE
[match-case] Object of type "Callable" is not callable

### ISSUE THREAD

**Issue by randolf-scholz** [microsoft/pyright#7794]
**Describe the bug**

I encountered the error message `Object of type "Callable" is not callable` when writing a function that makes use of [`pandas.Timedelta`](https://pandas.pydata.org/docs/reference/api/pandas.Timedelta.html). The setting `useLibraryCodeForTypes = false` seems relevant.

**Code or Screenshots**

[pyright playground](https://pyright-play.net/?code=GYJw9gtgBAxmA28CmMAuBLMA7AzgOgEMAjGKdCABzBFSgGEDFjkAoUSKVATwvSwHMylarQAqPJADUCIFm3DQKBLABMCOIVRpRR5JCqTxUBOaKgBeHROkgAFACJR9gDRQiYAK6rzuiPsPGAJRyLAbAUPxIqAD6AG6MHki2YEQAVgBcOlAAPvSM8MxIANpFfKgAuq6ilVAAVK44qEgUmWWBUAC0AHw66SxQA1AQBKgwABZQKRn9g7Mw6kg6egZGBLbt6pwqfbO7gyBRHiBYWzN78ziLDExEyOtQm8BeMDt7uweoRydPWDC2jc12lAAMRQQAy5FB7NcCrckPYyBosGBaPMbqw3rAFlBoq8MQMQAR0JcrBQkABREDgOzAewAbymAF8EVAkSj8oV7MEgA)

```python
from collections.abc import Callable
from typing import TypeVar
from pandas import Timedelta

T = TypeVar("T", bound=Timedelta)

def get_value(obj: T | Callable[[int], T], *, step: int) -> T:
    match obj:
        case Timedelta() as td:
            return td
        case Callable() as func:
            return func(step)  # ❌ "Callable" is not callable
        case _:
            raise TypeError
```

Relevant config:

- `pandas==2.2.2`
- `pandas-stubs` **not** installed
- `pyright==1.1.360`
- `useLibraryCodeForTypes = false` set in `pyproject.toml`

**Comment by randolf-scholz** [comment]
Equivalent if-else code without match-case does not trigger the error:

```python
from collections.abc import Callable
from datetime import timedelta
from typing import TypeVar
from pandas import Timedelta

T = TypeVar("T", bound=Timedelta)

def get_value2(obj: T | Callable[[int], T], *, step: int) -> T:
    if isinstance(obj, Timedelta):
        return obj
    if isinstance(obj, Callable):
        return obj(step)  # ✅
    raise TypeError
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.361, which I just published.
